### PR TITLE
fix close modal after delete and error message issue in Oauth client [NHUB-528]

### PR DESCRIPTION
**Pull request by devketanpro**
## Checklist

- [ ] This pull request is not adding new forms that use redux
- [ ] This pull request is adding missing TypeScript types to modified code segments where it's easy to do so with confidence
- [ ] This pull request is replacing `lodash.get` with optional chaining for modified code segments

### CODE REVIEW

File: assets/oauth_clients/components/Clients.tsx [hunk 1]
```diff
@@ -58,6 +58,7 @@ class Clients extends React.Component<any, any> {
 
         if (confirm(gettext('Would you like to delete client: {{name}}', {name: this.props.clientToEdit.name}))) {
             this.props.deleteClient('clients');
+            this.props.cancelEdit();
```
Comment (MarkLark86) [inline assets/oauth_clients/components/Clients.tsx]: The `async` branch should be reserved for backend changes only. We're trying to limit the frontend changes to none at all if possible.

This also looks like a common issue in the `develop` branch, so this fix might be better off going there (so it's in v2.8 release as well)

File: assets/utils.tsx [hunk 2]
```diff
@@ -513,6 +513,8 @@ export function errorHandler(error: {errorData: any} | Response, dispatch?: any,
     if ('errorData' in error) {
         if (setError) {
             dispatch(setError(error.errorData));
+            const errorMessage = Object.values(error.errorData).join(', ');
```
Comment (MarkLark86) [inline assets/utils.tsx]: Why are you displaying a notification for the validation error? This isn't the way validation currently works. The error should be displayed in the form, directly underneath the input that caused the error.

Have a look at the Company form, if you try and create 2 companies with the same name

**Comment by MarkLark86** [comment]
@devketanpro The backend code seems fine

File: assets/utils.tsx (hunk 2 above)
Comment (devketanpro) [reply assets/utils.tsx]: yep, I checked. It seems to me that this is also core work. not related to async

File: assets/oauth_clients/components/Clients.tsx (hunk 1 above)
Comment (devketanpro) [reply assets/oauth_clients/components/Clients.tsx]: would be good, to create a separate ticket for these issues, As I checked, these issues is also reproducing on nra-develop

**Review by MarkLark86 — APPROVED** [review]

**Pull request merged**